app/src/main/java/com/github/studydistractor/sdp/account/User.kt

```diff
@@ -0,0 +1,177 @@
+                || !birthday.containsKey("month")
+                || !birthday.containsKey("year")
+                || birthday.getValue("day") == 0
+                || birthday.getValue("month") == 0
+                || birthday.getValue("year") == 0){
+                throw java.lang.IllegalArgumentException("Birthday has an invalid format")
+            }
+        }
+    }
+
+    companion object {
+        private val path = "Users"
+
+        private val databaseRef: DatabaseReference = FirebaseDatabase.getInstance().getReference(path)
+
+        fun of(
+            userId: String,
+            firstname: String,
+            lastname: String,
+            phone: String,
+            birthday: HashMap<String, Int>
+        ): User {
+            if (userId == "") {
+                throw IllegalArgumentException("Empty userId")
+            }
+            Check.checkBirthdayFormat(birthday)
+            Check.checkNameFormat("firstname", firstname)
+            Check.checkNameFormat("lastname", lastname)
+            Check.checkPhoneFormat(phone)
+
+            val history = hashMapOf<String, String>()
+            val user = User(userId, firstname, lastname, phone, birthday, history,0)
+
+            user.uploadData()
+            return user
+        }
+
+        //fun fetchUserFromDatabase(userId: String): User {
+        //            if (userId.isEmpty()){
+        //                throw java.lang.IllegalArgumentException("userId is Empty")
+        //            }
+        //
+        //            val result = mutableStateListOf<User>()
+        //            databaseRef.addListenerForSingleValueEvent(object : ValueEventListener {
+        //                override fun onDataChange(snapshot: DataSnapshot) {
+        //                    for(user in snapshot.children) {
+        //                        val userItem = user.getValue(User::class.java)
+        //                        if(userItem != null) {
+        //                            val id = userItem.getUserId()
+        //                            if (userId == id) {
+        //                                result.add(userItem)
+        //                            }
+        //                        }
+        //                    }
+        //                }
+        //
+        //                override fun onCancelled(error: DatabaseError) {
+        //                    Log.d("Firebase", "loadPost:onCancelled " + error.toException().toString())
+        //                }
+        //            })
+        //
+        //            if (result.isEmpty()){
+        //                throw Exception("User not found in Database")
+        //            }
+        //
+        //            return result[0]
+        //        }
+
+    }
+
+    fun uploadData() {
+        Check.checkBirthdayFormat(birthday)
+        Check.checkNameFormat("Firstname", firstname)
+        Check.checkNameFormat("Lastname", lastname)
+        Check.checkPhoneFormat(phone)
+
```

Comment:
It is not the work of the createUser method ? Maybe change the name to uploadUser.